core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala

```diff
@@ -553,26 +566,27 @@ class JDBCSourceAsColumnarStore(private var _connProperties: ConnectionPropertie
                 val shift = if (GemFireCacheImpl.hasNewOffHeap) 5 else 13
                 assert(iterator.hasNext)
                 var smallestBucket = iterator.next()
-                var minBucketSize = smallestBucket.getTotalBytes >> shift
+                var minBucketSize = getInProgressBucketSize(smallestBucket, shift)
                 while (iterator.hasNext) {
                   val bucket = iterator.next()
-                  val bucketSize = bucket.getTotalBytes >> shift
+                  val bucketSize = getInProgressBucketSize(bucket, shift)
                   if (bucketSize < minBucketSize ||
                       (bucketSize == minBucketSize && Random.nextBoolean())) {
                     smallestBucket = bucket
                     minBucketSize = bucketSize
                   }
                 }
-                smallestBucket.getId
+                // update the in-progress size of the chosen bucket
+                val batchSize = batchSizeInBytes
```

Comment:
Not required val.

Reply:
@rishitesh Oh forgot why this was added. The "batchSizeInBytes" is a call by name closure that is supposed to be evaluated only when required. With the val it will be unnecessarily evaluated twice. Adding this back.